llvm/lib/Transforms/InstCombine/InstCombineShifts.cpp

```diff
@@ -1284,6 +1284,13 @@ Instruction *InstCombinerImpl::visitLShr(BinaryOperator &I) {
     return NewSub;
   }
 
+  // Fold (X + Y) / 2 --> (X & Y) iff (X u<= 1) && (Y u<= 1)
+  if (match(Op0, m_Add(m_Value(X), m_Value(Y))) && match(Op1, m_One()) &&
+      computeKnownBits(X, 0, &I).countMaxActiveBits() <= 1 &&
+      computeKnownBits(Y, 0, &I).countMaxActiveBits() <= 1) {
+    return BinaryOperator::CreateAnd(X, Y);
+  }
```

Comment:
Drop braces here. See https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements.